app/main.py

```diff
@@ -67,25 +55,46 @@ def empty_directory():
             traceback.print_exc()
 
 
-@app.route('/upload', methods=['POST'])
-def upload():
+def output(response_type, message, download_link):
+    if download_link == '':
+        response = [
+            {
+                'type': response_type,
+                'message': message
+            }
+        ]
+    else:
+        response = [
+            {
+                'type': response_type,
+                'message': message,
+                'download_link': download_link
+            }
+        ]
+    return jsonify({'response': response})
+
+
+@app.route('/api/v1.0/generate_badges', methods=['POST'])
+def main_task():
     """
-    Function to upload the form data from the webpage
+    Function to recive the input data from the user, process and send ouput
     """
     empty_directory()
-    csv = request.form['csv'].strip()
-    img = request.form['img-default']
-    custom_font = request.form['custfont']
-    file = request.files['file']
+    csv = request.form.get('csv', '').strip()
+    img = request.form.get('img-default', '')
+    custom_font = request.form.get('custfont', '')
+
+    if 'file' in request.files:
+        file = request.files['file']
 
-    # If default background is selected
+    # img-default is specified
     if img != '':
         if (img == 'user_defined.png'):
             bg_color = request.form['bg_color']
             do_svg2png(img, 1, bg_color)
```

Comment:
@abishekvashok what happens if user does not provide bg_color. Exception will be raised. Please use request.form.get('bg_color') and check for empty string.